sklearn/datasets/tests/test_samples_generator.py

```diff
@@ -245,6 +246,58 @@ def test_make_blobs():
         assert_almost_equal((X[y == i] - ctr).std(), std, 1, "Unexpected std")
 
 
+def test_make_blobs_n_features_list():
+    n_samples = [50, 30, 20]
+    X, y = make_blobs(n_samples=n_samples, n_features=2, random_state=0)
+
+    assert_equal(X.shape, (sum(n_samples), 2), "X shape mismatch")
+    assert_equal(y.shape, (sum(n_samples),), "y shape mismatch")
```

Comment:
You can do these two lines and more with an assertion that bincount equals the input

Reply:
Sorry @jnothman but I didn't get the `assertion that bincount equals the input`.
What did you mean exactly?

Reply:
`assert all(np.bincount(y) == n_samples))`